R/server.R

```diff
@@ -590,6 +600,156 @@ shinyServer(function(input, output, session) {
+            
+            
+            observeEvent(input[[paste0("slip_days_edit_", policy_name)]], {
+                slip_days$edit <- policy_name
+                policy <- slip_days$table[[policy_name]]
+                
+                
+                showModal(edit_slip_days_modal)
+                updateTextInput(session, "slip_days_policy_name", value = policy$name)
+                updateNumericInput(session, "num_slip_days", value = policy$num_slip_days)
+                updateSelectInput(session, "slip_days_order", selected = policy$order)
+                updateSelectizeInput(session, "assignments", choices = assign$table$assignment, selected = policy$assignments)
+                #ignoreInIt is important for closure of modal to work properly!
+            }, ignoreInit = TRUE)
+        })
+    })
+    
+    #### -------------------------- DELETING SLIP DAYS POLICY ----------------------------####
+    
+    slip_days_to_be_deleted <- reactiveValues(policy = NULL)
+    
+    observe({
+        req(slip_days$table)
+        lapply(names(slip_days$table), function(policy_name) {
+            local({
+                # Localize the variables to ensure correct binding in the observer
+                local_policy_name <- policy_name
+                
+                observeEvent(input[[paste0("slip_days_delete_", local_policy_name)]], {
+                    matched_policy <- slip_days$table[[local_policy_name]]
+                    
+                    if (!is.null(matched_policy)) {
+                        showModal(confirm_delete_slip_days)
+                        slip_days_to_be_deleted$policy <- local_policy_name
+                    } else {
+                        showNotification("Please pick a valid slip days policy to delete.", type = "error")
+                    }
+                }, ignoreInit = TRUE)
+            })
+        })
+    })
+    
+    observeEvent(input$delete_slip_days_confirm, {
+        req(slip_days_to_be_deleted$policy)
+        
+        # Remove the policy from both slip_days$table and policy$slip_days
+        slip_days$table[[slip_days_to_be_deleted$policy]] <- NULL
+        policy$slip_days <- policy$slip_days[!names(policy$slip_days) %in% slip_days_to_be_deleted$policy]
+        
+        # Clear the reactive value -- otherwise delete confirmation shows up
+        #if you have deleted this slip day policy wit hteh same name before.
+        slip_days_to_be_deleted$policy <- NULL
+        
+        removeModal()
+        showNotification("Slip days policy deleted successfully.", type = "message")
+    }, ignoreInit = TRUE)
+    
+    
+    #### -------------------------- DISPLAY SLIP DAYS UI ----------------------------####
+    
+    
+    output$slip_days_ui <- renderUI({
+        req(slip_days$table)
+        createSlipDaysCards(slip_days$table)
+    })
+    
+
+    
+    
     #### -------------------------- GRADING ----------------------------####
     
     observe({
```

Comment:
for the grading code,
`policy <- list(categories = policy$categories, slip_days = policy$slip_days) `
even though we haven't updated the package, we should add it anyway (it shouldn't break the app)

Reply:
let's do this after the merge -- so I don't have to do more commits before the merge.